integration-tests/emulated/tests/collectives/collectives-polkadot/src/tests/fellowship_treasury.rs

```diff
@@ -0,0 +1,233 @@
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::*;
+use asset_hub_polkadot_runtime::xcm_config::LocationToAccountId as AssetHubLocationToAccountId;
+use emulated_integration_tests_common::accounts::ALICE;
+use frame_support::{
+	dispatch::RawOrigin, sp_runtime::traits::Dispatchable, traits::fungible::Inspect,
+};
+use polkadot_runtime::OriginCaller;
+use polkadot_runtime_common::impls::VersionedLocatableAsset;
+use polkadot_runtime_constants::currency::UNITS;
+use xcm_executor::traits::ConvertLocation;
+
+// Fund Fellowship Treasury from Polkadot Treasury and spend from Fellowship Treasury.
+#[test]
+fn fellowship_treasury_spend() {
+	// initial treasury balance on Asset Hub in DOTs.
+	let treasury_balance = 20_000_000 * UNITS;
+	// target fellowship balance on Asset Hub in DOTs.
+	let fellowship_treasury_balance = 1_000_000 * UNITS;
+	// fellowship first spend balance in DOTs.
+	let fellowship_spend_balance = 10_000 * UNITS;
+
+	let init_alice_balance = AssetHubPolkadot::execute_with(|| {
+		<<AssetHubPolkadot as AssetHubPolkadotPallet>::Balances as Inspect<_>>::balance(
+			&Polkadot::account_id_of(ALICE),
+		)
+	});
+
+	Polkadot::execute_with(|| {
+		type RuntimeEvent = <Polkadot as Chain>::RuntimeEvent;
+		type RuntimeCall = <Polkadot as Chain>::RuntimeCall;
+		type Runtime = <Polkadot as Chain>::Runtime;
+		type Balances = <Polkadot as PolkadotPallet>::Balances;
+		type Treasury = <Polkadot as PolkadotPallet>::Treasury;
+
+		// Fund Treasury account on Asset Hub with DOTs.
+
+		let root = <Polkadot as Chain>::RuntimeOrigin::root();
+		let treasury_account = Treasury::account_id();
+
+		// Mist assets to Treasury account on Relay Chain.
+		assert_ok!(Balances::force_set_balance(
+			root.clone(),
+			treasury_account.clone().into(),
+			treasury_balance * 2,
+		));
+
+		let native_asset = Location::here();
+		let asset_hub_location: Location = [Parachain(1000)].into();
+		let treasury_location_on_asset_hub: Location = (Parent, PalletInstance(19)).into();
+
+		let teleport_call = RuntimeCall::Utility(pallet_utility::Call::<Runtime>::dispatch_as {
+			as_origin: bx!(OriginCaller::system(RawOrigin::Signed(treasury_account))),
+			call: bx!(RuntimeCall::XcmPallet(pallet_xcm::Call::<Runtime>::teleport_assets {
+				dest: bx!(VersionedLocation::V4(asset_hub_location.clone())),
+				beneficiary: bx!(VersionedLocation::V4(treasury_location_on_asset_hub)),
```

Comment:
Since the implementation `Pay` trait which used to execute the spend uses xcm `TransferAsset` instruction, we can use the pallet's location from which the xcm-executor will derive the account id.